src/lib/keyseq.ts

```diff
@@ -473,37 +475,44 @@ function numericPrefixToExstrSuffix(numericPrefix: KeyEventLike[]) {
  * translation map must be length-1 strings.
  */
 export function translateKeysUsingKeyTranslateMap(
-    keyEvents: KeyEventLike[],
+    keyEvents: MinimalKey[],
     keytranslatemap: { [inkey: string]: string },
 ) {
     for (let index = 0; index < keyEvents.length; index++) {
         const keyEvent = keyEvents[index]
         const newkey = keytranslatemap[keyEvent.key]
 
-        // KeyboardEvents can't have been translated, MinimalKeys may
-        // have been. We can't add anything to the MinimalKey without
-        // breaking a ton of other stuff, so instead we'll just assume
-        // that the only way we've gotten a MinimalKey is if the key
-        // has already been translated. We err way on the side of
-        // safety becase translating anything more than once would
+        // Translating anything more than once would
         // almost certainly mean oscillations and other super-weird
         // breakage.
-        const neverTranslated = keyEvent instanceof KeyboardEvent
-        if (neverTranslated && newkey !== undefined) {
-            // We can't update the keyEvent in place. However, the
-            // entire pipeline works with MinimalKeys all the way
-            // through, so we just swap the key event out for a new
-            // MinimalKey with the right key and modifiers copied from
-            // the original.
-            keyEvents[index] = new MinimalKey(newkey, {
-                altKey: keyEvent.altKey,
-                ctrlKey: keyEvent.ctrlKey,
-                metaKey: keyEvent.metaKey,
-                shiftKey: keyEvent.shiftKey,
+        if (!keyEvent.translated && newkey !== undefined) {
+            keyEvents[index] = minimalKeyFromKeyboardEventLike({
+                ...keyEvent,
+                key: newkey,
             })
+            keyEvents[index].translated = true
         }
     }
 }
+/**
+ * Convert keyboardEvent to internal type MinimalKey
+ * for further use. Key is obtained through layout-independent
+ * code if config says so.
+ */
+export function minimalKeyFromKeyboardEventLike(keyEvent: {
```

Comment:
Why not accept keyboardEvent? We need event.code and in future we'll probably want more context from the actual event

Reply:
it was so it could be used on line 489. KeyboardEvent has lots and lots of properties that aren't used in the function, I didn't see why it should be restricted :)